core/app/models/spree/stock/quantifier.rb

```diff
@@ -5,12 +5,13 @@ class Quantifier
 
       def initialize(variant)
         @variant = variant
-        @stock_items = Spree::StockItem.joins(:stock_location).where(variant_id: @variant, Spree::StockLocation.table_name => { active: true })
+        @stock_items = Spree::StockItem.joins(:stock_location).
+                         where(variant_id: @variant, Spree::StockLocation.table_name => { active: true })
       end
 
       def total_on_hand
         if variant.should_track_inventory?
-          stock_items.sum(:count_on_hand)
+          stock_items.map(&:count_on_hand).sum
```

Comment:
@aishmita-aggarwal The `map` call here would actually eager load all records. The original call would run the calculations more efficiently. I don't see how this pull request improves the code.